src/main.jsx

```diff
@@ -1,33 +1,37 @@
-import { StrictMode } from "react";
-import { createRoot } from "react-dom/client";
-import { BrowserRouter, Routes, Route } from "react-router-dom";
-import MainTemplate from "@/templates/MainTemplate";
-import Items from "@/pages/Items.jsx";
-import AddItem from "@/pages/AddItem.jsx";
-import Login from "@/pages/Login.jsx";
-import Signup from "@/pages/Signup.jsx";
-import NotFound from "@/pages/NotFound.jsx";
-import Home from "@/pages/Home.jsx";
-import Privacy from "@/pages/Privacy.jsx";
-import Faq from "@/pages/Faq.jsx";
-import Community from "@/pages/Community";
+import { StrictMode } from 'react';
+import { createRoot } from 'react-dom/client';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { ThemeProvider } from 'styled-components';
 
-createRoot(document.getElementById("root")).render(
+import AddItem from '@/pages/AddItem';
+import Community from '@/pages/Community';
+import Faq from '@/pages/Faq.jsx';
+import Home from '@/pages/Home.jsx';
+import Items from '@/pages/Items';
+import Login from '@/pages/Login.jsx';
+import NotFound from '@/pages/NotFound.jsx';
+import Privacy from '@/pages/Privacy.jsx';
+import Signup from '@/pages/Signup.jsx';
+import { GlobalStyle } from '@/styles/global';
+import theme from '@/styles/theme';
+
+createRoot(document.getElementById('root')).render(
   <StrictMode>
-    <BrowserRouter>
-      <Routes>
-        <Route path="/" element={<MainTemplate />}>
-          <Route index element={<Home />} />
-          <Route path="items" element={<Items />} />
-          <Route path="additem" element={<AddItem />} />
-          <Route path="login" element={<Login />} />
-          <Route path="signup" element={<Signup />} />
-          <Route path="privacy" element={<Privacy />} />
-          <Route path="community" element={<Community />} />
-          <Route path="faq" element={<Faq />} />
-          <Route path="*" element={<NotFound />} />
-        </Route>
-      </Routes>
-    </BrowserRouter>
+    <ThemeProvider theme={theme}>
+      <GlobalStyle />
+      <BrowserRouter>
+        <Routes>
+          <Route path='/' index element={<Home />} />
+          <Route path='items' element={<Items />} />
+          <Route path='additem' element={<AddItem />} />
```

Comment:
`Header`와 같은 컴포넌트를 각 필요한 페이지 컴포넌트에서 import해서 사용하고 있어요.
아래처럼 라우트 레벨에서 관리하는 걸로 개선해보는 것은 어떨까요? 참고해 볼 문서도 첨부드릴게요!

```suggestion
          <Route path='/' element={<MainLayout />}>
            <Route index element={<Home />} />
            <Route path='items' element={<Items />} />
            <Route path='additem' element={<AddItem />} />
          </Route>
```

- https://reactrouter.com/start/declarative/routing#nested-routes